addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,687 @@
+
+    public Tech getTech() {
+        // TODO change in Tech.MongoDB as soon as available
+        return Tech.Db;
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        // TODO change in Tech.MongoDB as soon as available
+        return technologies.includes(Tech.Db);
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INJECTION;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+    }
+
+    @Override
+    public String[] getDependency() {
+        return new String[] {};
+    }
+
+    public String getExtraInfo(String attack) {
+        return Constant.messages.getString(MESSAGE_PREFIX + "extrainfo." + attack);
+    }
+
+    @Override
+    public void init() {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Initialising MongoDB penertration tests");
+        }
+        if (this.getAttackStrength() == AttackStrength.LOW) {
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 2000;
+            doCrashScan = false;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = false;
+            doAuthBypass = true;
+        } else if (this.getAttackStrength() == AttackStrength.MEDIUM) {
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 3000;
+            doCrashScan = true;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = true;
+
+            doAuthBypass = true;
+        } else if (this.getAttackStrength() == AttackStrength.INSANE) {
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 3000;
+            doCrashScan = true;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = true;
+            doAuthBypass = true;
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, NameValuePair originalParam) {
+        isJsonPayload = originalParam.getType() == NameValuePair.TYPE_JSON;
+        super.scan(msg, originalParam);
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        boolean isBingo = false;
+        List<Integer> rtt = new ArrayList<>();
+        HttpMessage msgInjAttack, counterProofMsg;
+        String bodyInjAttack,
+                bodyCounterProof,
+                bodyBase = getBaseMsg().getResponseBody().toString();
+
+        if (!inScope(getTech())) {
```

Comment:
This check shouldn't be needed (and the `getTech()` method), core already ignores rules that don't target a given tech.